catalog/ui/search-ui/standard/src/main/java/org/codice/ddf/ui/searchui/standard/properties/ConfigurationStore.java

```diff
@@ -266,21 +262,71 @@ public void setTerrainProvider(String terrainProvider) {
     }
 
     private void setProxiesForImagery(List<String> imageryProviders) {
+        List<Map<String, Object>> newImageryProviders = imageryProviders.stream()
+                .map(this::getProviderConfigMap)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        List<Map<String, Object>> imageryProvidersToStop = new ArrayList<>();
+        List<Map<String, Object>> imageryProvidersToStart = new ArrayList<>();
+
+        findDifferences(proxiedImageryProviders, newImageryProviders, imageryProvidersToStart);
+        findDifferences(newImageryProviders, proxiedImageryProviders, imageryProvidersToStop);
+
+        List<String> proxiesToStop = imageryProvidersToStop.stream()
+                .map(provider -> urlToProxyMap.get(provider))
+                .collect(Collectors.toList());
+
+        stopImageryEndpoints(proxiesToStop);
+        startImageryEndpoints(imageryProvidersToStart);
+        proxiedImageryProviders = newImageryProviders;
+    }
+
+    private void findDifferences(List<Map<String, Object>> innerList,
+            List<Map<String, Object>> outerList,
+            List<Map<String, Object>> differences) {
+        for (Map<String, Object> newProvider : outerList) {
+            boolean found = false;
+            for (Map<String, Object> provider : innerList) {
+                if (provider.equals(newProvider)) {
+                    found = true;
+                }
+            }
+            if (!found) {
+                differences.add(newProvider);
+            }
+        }
+    }
+
+    private void stopImageryEndpoints(List<String> imageryEndpoints) {
         if (imageryEndpoints.size() > 0) {
             for (String endpoint : imageryEndpoints) {
                 try {
                     httpProxy.stop(endpoint);
+                    urlToProxyMap.remove(endpoint);
+                    imageryEndpoints.remove(endpoint);
```

Comment:
This will probably always work, but I hate modifications to a collection as you're walking it. Better to switch to an explicit iterator for the loop and call remove on that as it's a more correct and safe idiom.
```java
for (Iterator<String> iterator = imageryEndpoints.iterator(); iterator.hasNext(); ) {
    String endpoint = iterator.next();
    try {
        httpProxy.stop(endpoint);
        urlToProxyMap.remove(endpoint);
        iterator.remove();
    } catch (Exception e) {
        LOGGER.error("Unable to stop proxy endpoint: {}", endpoint, e);
    }
}
```